app/models/link.rb

```diff
@@ -4,6 +4,7 @@ class Link < Datafile
   # Some legacy datafiles have invalid dates (e.g. 31/06/15).
   # When this occurs the date attributes are empty, therefore we cannot invoke this call-back
   before_save :set_end_date, unless: ->{ year.nil? }
+  before_save :set_start_date, unless: ->{ year.nil? }
```

Comment:
@hannako How about a `set_date_range` method that computes both at once? You'd avoid passing around `date_type` as arguments and would get rid of the if statements.

Maybe something like:

```
def set_date_range
   [self.start_date, self.end_date] = compute_date_range
end

def compute_date_range
   return monthly_date_range          if dataset.monthly?
end

def monthly_date_range
   [Date.new(year.to_i, month.to_i).beginning_of_month, Date.new(year.to_i, month.to_i).end_of_month]
end
```

Reply:
That would indeed look slightly better. However, since the plan is to remove start_date altogether (which we haven't yet for the reasons explained above) it'll be easier, and diff will be clearer. Make sense, @laurentqro ?

Reply:
👍